vars/piperExecuteBin.groovy

```diff
@@ -171,17 +172,18 @@ void credentialWrapper(config, List credentialInfo, body) {
                 credentialsId = config[cred.id]
             }
             if (credentialsId) {
-                switch(cred.type) {
-                    case "file":
+                switch("${cred.type}") {
```

Comment:
oups, wouldn't be compatible, right? so drop the comment.

Reply:
But then we have to make sure that _type: "file"_ is working as well! So we need a test like

`testPiperExecuteBinCredentialsSetAsEnumString()`

Reply:
Can someone explain the backward-compatibility concerns? Inside our code base we should be able to identify all occurrences and to do a shift accordingly. Do we have some cases where our code base is incorporated into other libraries which might not work afterwards anymore? Are there any contracts? I would say when someone re-uses the lib here it is the duty of that user to do a proper inbound qualification and to adjust accordingly. For such re-use cases it would be beneficials to define APIs in order to create awareness what must be kept stable. Without that basically everyhting needs to be kept stable - which is not a that cool working mode.

Reply:
I guess we have some consumers of PiperExecuteBin outside the Piper OS, running

`piperExecuteBin(..., [[type: "file",...]])
`

Reply:
This was my assumption too. @OliverNocon [mentioned](https://github.com/SAP/jenkins-library/issues/2121#issuecomment-704465862) in the corresponding issue #2121 that piperExecuteBin is consumed outside of this library, maybe ha can give further information on who exactly does that. 

I can only think of the [S4 SDK lib](https://github.com/SAP/cloud-s4-sdk-pipeline-lib), which is now however moved into this lib. But it was still a separate lib when Oliver wrote his comment.

Reply:
> But then we have to make sure that _type: "file"_ is working as well! So we need a test like
> 
> `testPiperExecuteBinCredentialsSetAsEnumString()`

done